js/components/frame.js

```diff
@@ -452,7 +452,10 @@ class Frame extends ImmutableComponent {
         break
       case 'view-source':
         const sourceLocation = UrlUtil.getViewSourceUrlFromUrl(this.webview.getURL())
-        windowActions.newFrame({location: sourceLocation}, true)
+        let src = this.frame.get('src')
+        if (UrlUtil.isHttpAddress(src) && !UrlUtil.isImageAddress(src)) {
```

Comment:
can we split this change out from the shortcut? I think we probably want to make these extra checks in `getViewSourceUrlFromUrl` and maybe return a null location if there is not valid view-source for the url
@bradleyrichter is this shortcut in your list?